### PR TITLE
#159522903 Fix create new category bug

### PR DESCRIPTION
**Pull request by kevinejiro**
**What does this PR do?**

This PR ensures that an attribute type is saved properly when Adding an asset category.

**Description of Task to be completed?**

- Update Checkbox.test.tsx
- Update CreateCategoryPage.test.tsx
- Update CreateCategoryPage

**How should this be manually tested?**

xxx

**Any background context you want to provide?**

xxx

**What are the relevant pivotal tracker stories?**

[#159522903](https://www.pivotaltracker.com/n/projects/2170023/stories/159522903)

**Screenshots (if appropriate)**

xxx

**Questions:**

xxx

### CODE REVIEW

**Comment by FelixWambiri** [comment]
Please address the conflicts.

**Comment by Dammyy** [comment]
@kevinejiro please resolve the merge conflict.

File: src/components/Attribute/index.tsx [hunk 1]
```diff
@@ -126,8 +126,18 @@ class Attribute extends React.Component<AttributeProps> {
       error = 'This option already exists.';
       return error;
     }
-
-    this.setState({ fieldOptions: this.state.fieldOptions.concat(option) });
+    const fieldOptions = [
+      ...this.state.fieldOptions,
+      option,
+    ];
+    this.setState({ fieldOptions });
```
Comment (andela-kakpobome) [inline src/components/Attribute/index.tsx]: Why the change from concat to destructure?

File: src/components/Attribute/fixtures.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export const inputTypes = [
-  { id: '1', name: 'dropdown' },
-  { id: '2', name: 'textarea' },
-  { id: '3', name: 'radio button' },
-  { id: '4', name: 'text' },
+  { id: '1', name: 'Text' },
+  { id: '2', name: 'Radio button' },
+  { id: '3', name: 'Textarea' },
+  { id: '4', name: 'Checkbox' },
```
Comment (andela-kakpobome) [inline src/components/Attribute/fixtures.ts]: Why capitalize all these and also remove the dropdown option?

File: src/components/Attribute/index.tsx [hunk 3]
```diff
@@ -214,7 +232,11 @@ class Attribute extends React.Component<AttributeProps> {
   }
 
   render() {
-    const allowedSelectOptions: any = ['dropdown', 'textarea', 'radio button', 'text'];
+    const allowedSelectOptions: any = [
+      'Dropdown',
+      'Radio button',
+      'Checkbox',
+    ];
```
Comment (andela-kakpobome) [inline src/components/Attribute/index.tsx]: Why this change?

File: src/components/Attribute/interfaces.d.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ interface AttributeValue {
   label?: string;
   placeholder?: string;
   isDisabled?: boolean;
```
Comment (andela-kakpobome) [inline src/components/Attribute/interfaces.d.ts]: I know you didn't add this, but it shouldn't be here, it's a duplicate of `10`

File: src/components/CheckBox/interfaces.d.ts [hunk 5]
```diff
@@ -1,6 +1,10 @@
 export interface CheckBoxProps {
-  name: string;
+  name?: string;
   makeActive: () => void;
+  onAttributesChange?: (event: object) => void;
   isActive: boolean;
   toolTip?: string;
+  label?: string;
+  inputName?: string;
+  onChecked: (event: object) => void;
```
Comment (andela-kakpobome) [inline src/components/CheckBox/interfaces.d.ts]: 1. What is the difference between name, input name and label?
2. What's the value of `onAttributeChange`
3. What's the difference between `makeActive` and `onChecked`? Why can't just one of those be used?

File: src/pages/CategoryPageForm/CategoryPageForm.test.tsx [hunk 6]
```diff
@@ -70,7 +70,7 @@ describe('The CategoryPageForm component', () => {
 
     wrapper.find('.additional-fields__new-attribute').first().simulate('click');
     expect(wrapper.instance().state.category.additionalAttributes).toEqual([{
-      inputControl: 'text',
+      inputControl: 'Text',
```
Comment (andela-kakpobome) [inline src/pages/CategoryPageForm/CategoryPageForm.test.tsx]: Why are you capitalising?

File: src/pages/CategoryPageForm/interfaces.d.ts [hunk 7]
```diff
@@ -27,11 +27,13 @@ export interface CategoryPageFormProps {
 }
 
 export interface Errors {
-  newCategoryName: string ;
+  newCategoryName: string;
 }
 
 interface Attributes {
   label?: string;
```
Comment (andela-kakpobome) [inline src/pages/CategoryPageForm/interfaces.d.ts]: How come `label` is optional though? Is there a scenario where we don't need it?

File: src/pages/CategoryPageForm/interfaces.d.ts [hunk 8]
```diff
@@ -27,11 +27,13 @@ export interface CategoryPageFormProps {
...
   inputControl: string;
   isRequired: boolean;
+  placeholder: string;
+  choices: any[];
```
Comment (andela-kakpobome) [inline src/pages/CategoryPageForm/interfaces.d.ts]: Not all attributes would be dropdowns, checkboxes or radio buttons, so this cannot be compulsory

File: src/pages/CategoryPageForm/index.tsx [hunk 9]
```diff
@@ -278,7 +277,7 @@ export class CategoryPageForm extends React.Component<CategoryPageFormProps, Cat
             ...prevState.category.additionalAttributes,
             {
               label: '',
-              inputControl: 'text',
+              inputControl: 'Text',
```
Comment (andela-kakpobome) [inline src/pages/CategoryPageForm/index.tsx]: I don't understand this capitalization effort, please explain

File: src/pages/CategoryPageForm/index.tsx [hunk 10]
```diff
@@ -343,7 +342,7 @@ export class CategoryPageForm extends React.Component<CategoryPageFormProps, Cat
         return index === indexInState
           ? {
             ...attribute,
-            label: event.target.value,
+            [event.target.name]: event.target.value,
```
Comment (andela-kakpobome) [inline src/pages/CategoryPageForm/index.tsx]: Since the `label` is the value we want to set, what's the value of this change?

File: src/components/Attribute/index.tsx (hunk 1 above)
Comment (kevinejiro) [reply src/components/Attribute/index.tsx]: This was just `Syntactic Sugar`. They do exactly the same thing, but i could change it back if you recommend it.

File: src/components/Attribute/index.tsx (hunk 3 above)
Comment (kevinejiro) [reply src/components/Attribute/index.tsx]: I believed textarea and text should not have options.

File: src/components/Attribute/index.tsx (hunk 1 above)
Comment (kevinejiro) [reply src/components/Attribute/index.tsx]: we are also using the variable `field options ` on line `137`.

**Comment by andela-kakpobome** [comment]
Fix conflicts here

File: src/pages/CategoryPageForm/index.tsx (hunk 10 above)
Comment (kevinejiro) [reply src/pages/CategoryPageForm/index.tsx]: The previous version only changes the value of the `label` which means the rest (`inputControl`, `isRequired ` and `choices` ) would still be their default value when creating an asset category, but with this it handles change for all the aforementioned.

File: src/pages/CategoryPageForm/interfaces.d.ts (hunk 8 above)
Comment (kevinejiro) [reply src/pages/CategoryPageForm/interfaces.d.ts]: This is true. Thanks for pointing it out

File: src/components/AddCategoryOptions/index.tsx [hunk 11]
```diff
@@ -62,7 +62,7 @@ class AddCategoryOptions extends React.Component<AddCategoryOptionsProps> {
     return(
       <div className="add-option-input-field" onKeyPress={this.onKeyPress}>
         <InputBox
-          name="option"
+          name="choice"
```
Comment (andela-kakpobome) [inline src/components/AddCategoryOptions/index.tsx]: What's the value of this change?
